commit 524f363cb1588e3b267044531fdf051bc204e669 (refs/changes/22/822/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-31T20:42:20+00:00 (2 years, 11 months ago)
    
    DOCKER-959 Unable to pull from registry when registry response sends multiple Docker-Distribution-Api-Version headers
    IMGAPI-600 AdminImportDockerImage 'error' progress message should use restify-errors err.name as a fallback code
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 6c8e57f..8445b24 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,8 @@
 # IMGAPI changelog
 
+## 3.1.3
+
+
 ## 3.1.2
 
 - DOCKER-950 docker pull fails on registry that does not use authentication
diff --git a/lib/images.js b/lib/images.js
index 9b5e4e8..03ced87 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -3562,9 +3562,10 @@ function apiAdminImportDockerImage(req, res, callback) {
                     /*
                      * `restify.RestError` instances will have `err.code`. More
                      * vanilla `restify.HttpError` instances may have
-                     * `err.body.code`.
+                     * `err.body.code` (if the response body was JSON with a
+                     * "code"), else the contructor name is a code.
                      */
-                    code: err.code || (err.body && err.body.code),
+                    code: err.code || (err.body && err.body.code) || err.name,
                     message: err.message
                 }
             });
diff --git a/package.json b/package.json
index 8f4b729..69c551a 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "3.1.2",
+  "version": "3.1.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -10,7 +10,7 @@
     "bunyan": "1.8.1",
     "cmdln": "3.2.1",
     "dashdash": "1.10.0",
-    "docker-registry-client": "3.2.3",
+    "docker-registry-client": "3.2.4",
     "effluent-logger": "git+https://github.com/joshwilsdon/effluent-logger.git#d662f161a07f94045ad2afb45442931511c40e51",
     "expiring-lru-cache": "2.1.0",
     "extsprintf": "1.2.0",
